components/brave_wallet/browser/fil_tx_manager.cc

```diff
@@ -224,6 +228,53 @@ FilTxStateManager* FilTxManager::GetFilTxStateManager() {
   return static_cast<FilTxStateManager*>(tx_state_manager_.get());
 }
 
-void FilTxManager::UpdatePendingTransactions() {}
+void FilTxManager::UpdatePendingTransactions() {
+  auto pending_transactions = tx_state_manager_->GetTransactionsByStatus(
+      mojom::TransactionStatus::Submitted, absl::nullopt);
+  for (const auto& pending_transaction : pending_transactions) {
+    auto cid = pending_transaction->tx_hash();
+    uint64_t seconds =
+        (base::Time::Now() - pending_transaction->submitted_time()).InSeconds();
+    // StateSearchMsgLimited looks back up to limit epochs in the chain for a
+    // message. Returns its receipt and the tipset where it was executed.
+    // We assume that 1 block mined per second and taking a
+    // difference between current time and transaction submission to calculate
+    // the limit. In case of zero we are looking at last message.
+    uint64_t limit_epochs = seconds == 0 ? 1 : seconds;
+    json_rpc_service_->GetFilStateSearchMsgLimited(
+        cid, limit_epochs,
+        base::BindOnce(&FilTxManager::OnGetFilStateSearchMsgLimited,
+                       weak_factory_.GetWeakPtr(), pending_transaction->id()));
+  }
+  known_no_pending_tx_ = pending_transactions.empty();
+  CheckIfBlockTrackerShouldRun();
+}
+
+void FilTxManager::OnGetFilStateSearchMsgLimited(
+    const std::string& tx_meta_id,
+    int64_t exit_code,
+    mojom::FilecoinProviderError error,
+    const std::string& error_message) {
+  if (error != mojom::FilecoinProviderError::kSuccess)
+    return;
+  std::unique_ptr<FilTxMeta> meta =
+      GetFilTxStateManager()->GetFilTx(tx_meta_id);
+  if (!meta)
+    return;
+  mojom::TransactionStatus status = (exit_code == 0)
+                                        ? mojom::TransactionStatus::Confirmed
+                                        : mojom::TransactionStatus::Error;
+  meta->set_status(status);
+  meta->set_confirmed_time(base::Time::Now());
```

Comment:
I think currently for other coin types, we only do `set_confirmed_time` if it's confirmed and not error.

Reply:
done